### PR TITLE
Let initial_layout be a list

### LINKED ISSUE #735: Let initial_layout be a list
### What is the expected enhancement?
Lets the user pass a list of numbers as an `initial_layout` to the `compile` function that maps the circuit from top to bottom in the quantum registers.

### PR DESCRIPTION
**Pull request by nonhermitian**
Fix #735 

### Summary

This PR lets the user pass a list of numbers as an `initial_layout` to the `compile` function that maps the circuit from top to bottom in the quantum registers.

### Details and comments

Passing a custom `initial_layout` is a headache, when it should be a simple process.  This PR makes it simple.

### CODE REVIEW

File: qiskit/transpiler/_transpiler.py [hunk 1]
```diff
@@ -343,6 +348,35 @@ def _best_subset(backend, n_qubits):
     return best_map
 
 
+def layout_as_list(layout_list, qregs):
```
Comment (ajavadia) [inline qiskit/transpiler/_transpiler.py]: should be called `layout_as_dict`? (since it returns dict)

**Comment by ajavadia** [comment]
can you add a test for doing this on a circuit with 2 different registers?
also can you add a line to the changelog?

**Comment by diego-plan9** [comment]
Do we really need  to support both formats in `qiskit.transpiler.compile()`? It is usually beneficial to have "just one way of doing things" in the core functionality, and avoid ambiguity and polymorphic arguments to the extent possible. If the change is mostly for user convenience, could it be handled just at the wrapper level, ensuring `qiskit.transpiler` and the rest of the codebase deal with just one representation (either one is fine - the key point is making it unambiguous)?

**Comment by nonhermitian** [comment]
Changed the name to hopefully satisfy @ajavadia.  The current test is already over three different named registers.

@diego-plan9 it is for user simplicity, but it makes sense to have it at all levels because it makes low-level users life better as well.  I don't really see an issue here.  It is somewhat akin to lists and np.arrays.  It is common to have `np.asarray(list)` instead of demanding a NumPy array is passed.  Here we just do a list->dict conversion if necessary.

**Comment by diego-plan9** [comment]
@nonhermitian it's fine! Changes along those lines (modify the signature of a rather important function, and in essence for accommodating user needs) can be more tricky than they seem and have implications that might bite us in the future, and it's important to make the rationale explicit and put as many eyes on them as possible: hence the question.

**Comment by nonhermitian** [comment]
Closing as we have discussed an alternative implementation.

**Pull request closed without merging**